packages/react/src/components/Table/TableColumnCustomizationModal/TableColumnCustomizationModal.story.jsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState } from 'react';
+import { action } from '@storybook/addon-actions';
+import { boolean, object, text } from '@storybook/addon-knobs';
+
+import { DragAndDrop } from '../../../utils/DragAndDropUtils';
+import StoryNotice, { experimentalStoryTitle } from '../../../internal/StoryNotice';
+
+import TableColumnCustomizationModal from './TableColumnCustomizationModal';
+
+const getColumns = () => [
+  {
+    id: 'string',
+    name: 'String',
+  },
+  {
+    id: 'date',
+    name: 'Date',
+  },
+  {
+    id: 'select',
+    name: 'Select',
+  },
+  {
+    id: 'secretField',
+    name: 'Secret Information',
+  },
+  {
+    id: 'status',
+    name: 'Status',
+  },
+  {
+    id: 'number',
+    name: 'Number',
+  },
+  {
+    id: 'boolean',
+    name: 'Boolean',
+  },
+  {
+    id: 'node',
+    name: 'React Node',
+  },
+  {
+    id: 'object',
+    name: 'Object Id',
+    renderDataFunction: ({ value }) => {
+      return value?.id;
+    },
+  },
+];
+
+const getOrdering = () => [
+  { columnId: 'string' },
+  { columnId: 'date' },
+  { columnId: 'select' },
+  { columnId: 'secretField', isHidden: true },
+  { columnId: 'status' },
+  { columnId: 'number' },
+];
+
+const getColumnGroupMapping = () => [
+  { id: 'groupA', name: 'Group A', columnIds: ['date', 'select'] },
+  { id: 'groupB', name: 'Group B', columnIds: ['secretField', 'status', 'number', 'boolean'] },
+];
+
+export const Experimental = () => (
+  <StoryNotice componentName="TableColumnCustomizationModal" experimental />
+);
+Experimental.storyName = experimentalStoryTitle;
+
+export default {
+  title: '1 - Watson IoT/Table/☢️ TableColumnCustomizationModal',
+
+  parameters: {
+    component: TableColumnCustomizationModal,
+  },
+};
+
+export const PlaygroundWithKnobs = () => {
+  const hasVisibilityToggle = boolean('hasVisibilityToggle', false);
+  const demoGroupMapping = boolean('demo column groups', false);
+
+  return (
+    <TableColumnCustomizationModal
+      key={`table-column-customization-modal${hasVisibilityToggle}}`} // Make sure component reloaded
+      groupMapping={demoGroupMapping ? object('groupMapping', getColumnGroupMapping()) : []}
+      hasVisibilityToggle={hasVisibilityToggle}
+      availableColumns={object('availableColumns', getColumns())}
+      initialOrdering={object('initialOrdering', getOrdering())}
+      onClose={action('onClose')}
+      onChange={action('onChange')}
+      onReset={action('onReset')}
+      onSave={action('onSave')}
+      open={boolean('open', true)}
+      pinnedColumnId={text('pinnedColumnId', '')}
```

Comment:
We might need to move this up into a variable. Not sure what's happening. If I add the "object" column, drag it to the top, and then set it as the pinned id, it resets the state of the modal and marks the object column as unchecked and unselectable.

![reset](https://user-images.githubusercontent.com/609466/142484843-303b57d4-cb2d-4278-90a0-32628344818c.gif)

Reply:
Indeed this doesn't demo well. I'm going to replace the pinnedColumnId text knob with a boolean "demo pinned column" since there are some restrictions on using setting the pinned column as described in the documentation:
"The pinned column must be present in the availableColumns prop and it must be the first column in the initialOrdering prop. It is not possible to pin multiple columns."